views/NewChart.react.js

```diff
@@ -1,9 +1,107 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+import { compose } from 'redux'
+import { withStyles } from '@material-ui/core/styles'
+import Button from '@material-ui/core/Button'
+import { connect } from 'react-redux'
+
+import Header from './components/Header'
+import Sidebar from './components/Sidebar'
+import Form from './components/Form'
+
+import BarChartFields from './forms/BarChartFields'
+
+import getCounts from './fe-utils/countUtil'
+import getAvatar from './fe-utils/avatarUtil'
+import { fetchSubjects, createChart } from './fe-utils/fetchUtil'
+import getDefaultStyles from './fe-utils/styleUtil'
+import { chartStyles } from './styles/chart_styles'
+
+const NewChart = ({ user, classes }) => {
+  const [formValues, setFormValues] = useState({})
+  const [fieldLabelValueMap, setFieldLabelValueMap] = useState([])
```

Comment:
I think it's worth bringing this state and the <Form> below to a `./forms/ChartForm` component. This will allow the view component to remove some of its logic and it can pass the `onSubmit` prop to the form to handle form submission to the API. What do you think about this approach?

Reply:
[8a7c3d0](https://github.com/AMP-SCZ/dpdash/pull/84/commits/8a7c3d014d0266427e76ae9b5d15d0618767ac3a)